webrender/src/render_backend.rs

```diff
@@ -470,6 +481,38 @@ impl RenderBackend {
         let mut notifier = self.notifier.lock();
         notifier.as_mut().unwrap().as_mut().unwrap().new_scroll_frame_ready(composite_needed);
     }
+
+    fn handle_vr_compositor_command(&mut self, ctx_id: WebGLContextId, cmd: VRCompositorCommand) {
+        match cmd {
+            VRCompositorCommand::Create(compositor_id) => {
+                let mut creator = self.vr_compositor_creator.lock();
+                let compositor = creator.as_mut().unwrap().as_mut().unwrap().create_compositor(compositor_id);
+                if let Some(compositor) = compositor {
+                    self.vr_compositors.insert(compositor_id, compositor);
+                }
+            }
+            VRCompositorCommand::SyncPoses(compositor_id, near, far, sender) => {
+                if let Some(ref compositor) = self.vr_compositors.get(&compositor_id) {
+                    let pose = compositor.sync_poses(near, far);
+                    let _result = sender.send(Ok(pose));
+                } else {
+                    let _result = sender.send(Err(()));
+                }
+            }
+            VRCompositorCommand::SubmitFrame(compositor_id, left_bounds, right_bounds) => {
+                if let Some(ref compositor) = self.vr_compositors.get(&compositor_id) {
+                    let texture = self.resource_cache.get_webgl_texture(&ctx_id);
```

Comment:
The whole VR logic seems to be mostly independent of the WR, except for this `resource_cache` access. Do the compositors and their compositor creator have to live within the WR backend/renderer at all? Perhaps, we could narrow down the interface between those and reduce the integration.